### PR TITLE
[Feature]: add transition method to the Containment API

### PR DESCRIPTION
**Pull request by danyf90**
The transition method allows to replace the last child of the container with a new one, with the possibility to animate the change

**Why**
Allow to have transition animation within the Containment API

**Changes**
Add transition(to:,in:,duration:options:,completion:) method to ViewController extension

**Tasks**
* [x] Add relevant tests, if needed
* [x] Add documentation, if needed
* [ ] Update README, if needed
* [ ] Ensure that the demo project works properly

### CODE REVIEW

File: Tempura/Core/ViewController+Containment.swift [hunk 1]
```diff
@@ -16,10 +16,43 @@ extension ViewController {
...
-  
+
+  /// Replace a ViewController (and its rootView) to the last child of self.
+  /// You must provide a ContainerView inside self.rootView, the latest view of that container view
+  /// is replaced with the rootView of the child ViewController with a cross-dissolve transition
+  public func transition<View: ViewControllerModellableView>(
+    to child: ViewController<View>,
+    in view: ContainerView,
+    duration: Double = 0.3,
+    options: UIView.AnimationOptions = [],
```
Comment (AlfioRusso) [inline Tempura/Core/ViewController+Containment.swift]: As default I would use `.transitionCrossDissolve` but remember to use them below.

File: Tempura/Core/ViewController+Containment.swift [hunk 2]
```diff
@@ -16,10 +16,43 @@ extension ViewController {
...
+    UIView.transition(
+      from: lastView,
+      to: child.rootView,
+      duration: duration,
+      options: [.transitionCrossDissolve],
+      completion: { _ in
+        self.addChild(child)
+        child.didMove(toParent: self)
+        lastViewVC.willMove(toParent: nil)
+        lastViewVC.viewWillDisappear(false)
```
Comment (AlfioRusso) [inline Tempura/Core/ViewController+Containment.swift]: You should call the `will` methods before starting the animation.

File: Tempura/Core/ViewController+Containment.swift [hunk 3]
```diff
@@ -16,10 +16,43 @@ extension ViewController {
...
+    }
+
+    UIView.transition(
+      from: lastView,
+      to: child.rootView,
+      duration: duration,
+      options: [.transitionCrossDissolve],
+      completion: { _ in
+        self.addChild(child)
+        child.didMove(toParent: self)
```
Comment (AlfioRusso) [inline Tempura/Core/ViewController+Containment.swift]: Don't we need to actually add child as subview?

Comment (danyf90) [reply Tempura/Core/ViewController+Containment.swift]: It should be done by the UIView.transition: https://developer.apple.com/documentation/uikit/uiview/1622562-transition

```
fromView
The starting view for the transition. By default, this view is removed from its superview as part of the transition.

toView
The ending view for the transition. By default, this view is added to the superview of fromView as part of the transition.
```

Comment (AlfioRusso) [reply Tempura/Core/ViewController+Containment.swift]: Cool! 🙌

**Pull request merged**